autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java

```diff
@@ -51,4 +73,249 @@ public void contributeSecond() {
+    warningDisplayed = findWarningLink(wizardPage, controlDivId, false);
+    assertFalse(warningDisplayed);
+
+    // Add one file having duplicate and one link having too
+    wizardPage.addFile(3, FIRST_FILE_NAME);
+    wizardPage.addUrl(3, FIRST_LINK_NAME, FIRST_LINK_NAME);
+    warningDisplayed = findWarningLink(wizardPage, controlDivId, true);
+    assertTrue(warningDisplayed);
+    checkDuplicateItem(false, FIRST_FILE_NAME, FIRST_LINK_NAME);
+    wizardPage.get();
+    wizardPage.universalControl(3).deleteResource(FIRST_FILE_NAME);
+    warningDisplayed = findWarningLink(wizardPage, controlDivId, true);
+    assertTrue(warningDisplayed);
+    checkDuplicateItem(false, FIRST_LINK_NAME);
+    wizardPage.get();
+    wizardPage.universalControl(3).deleteResource(FIRST_LINK_NAME);
+    warningDisplayed = findWarningLink(wizardPage, controlDivId, false);
+    assertFalse(warningDisplayed);
+  }
+
+  private boolean findWarningLink(
+      WizardPageTab wizardPage, String controlDivId, boolean isWarningDisplayed) {
+    By linkTextXpath =
+        By.xpath(
+            "//div[@id='"
+                + controlDivId
+                + "']//a[text()="
+                + quoteXPath(DUPLICATE_WARNING_LINK_TEXT)
+                + "]");
+    if (isWarningDisplayed) {
+      WebElement warningLink =
+          wizardPage.getWaiter().until(ExpectedConditions.elementToBeClickable(linkTextXpath));
+      // Check if the warning link text is correct
+      Assert.assertEquals(
+          warningLink.getText(), "Click here to view existing items that contain duplicates");
+      warningLink.click();
+      return true;
+    } else {
+      wizardPage.getWaiter().until(ExpectedConditions.invisibilityOfElementLocated(linkTextXpath));
+    }
+    return false;
+  }
+
+  private void checkDuplicateItem(boolean isUniqueEnforced, String... duplicateContentTexts) {
+    DuplicatesTab duplicateDataPage = new DuplicatesTab(context).get();
+
+    for (String duplicateContentText : duplicateContentTexts) {
+      By duplicateContentXpath =
+          By.xpath(
+              "//div[starts-with(@class,'duplicate')]//*[contains(text(),"
+                  + quoteXPath(duplicateContentText)
+                  + ")]");
+      WebElement duplicateContent =
+          duplicateDataPage
+              .getWaiter()
+              .until(ExpectedConditions.visibilityOfElementLocated(duplicateContentXpath));
+      Assert.assertEquals(
+          duplicateContent.getText(), "'" + duplicateContentText + "' is also used by:");
+    }
+
+    By duplicatesFoundXpath = By.xpath("//a[text()='" + DUPLICATE_CHECK_ITEM_NAME + "']");
+    WebElement duplicateItem =
+        duplicateDataPage
+            .getWaiter()
+            .until(ExpectedConditions.visibilityOfElementLocated(duplicatesFoundXpath));
+    // Check if the correct duplicate item is found
+    Assert.assertEquals(duplicateItem.getText(), "uniquenessTest - Duplicate Check");
```

Comment:
You've got a constant for that String don't you?

```suggestion
    Assert.assertEquals(duplicateItem.getText(), DUPLICATE_CHECK_ITEM_NAME);
```